apps/bluetooth/app.js

```diff
@@ -1,289 +1,331 @@
-  var player = null
-  var uuid = (property.get('ro.boot.serialno') || '').substr(-6)
-  var productName = property.get('ro.rokid.build.productname') || 'Rokid-Me'
-  var name = [ productName, uuid ].join('-')
-  var nameToSpeak = [ productName, `<num=tel>${uuid}</num>` ].join('')
+  var a2dp = null
+  var deviceName = system.getDeviceName()
+  logger.debug(`deviceName = ${deviceName}`)
   var textTable = require('./texts.json')
-  var bluetoothMessage = {
-    'a2dpstate': 'closed',
-    'connect_state': 'invalid',
-    'connect_name': null,
-    'play_state': 'invalid',
-    'broadcast_state': 'closed'
-  }
-  var activityPlayState = false
-  var activityNlpIntent = null
   var BLUETOOTH_MUSIC_ID = 'RDDE53259D334860BA9E98CB3AB6C001'
+  var needResume = false
+  var lastIntent = null
 
-  function broadcast (subsequent) {
-    player.start(name, subsequent)
+  function getText (mode, label) {
+    var str = textTable[mode][label]
+    if (str === undefined || str === null) {
+      str = textTable[label]
+    }
+    return str
   }
 
-  function bluetoothMessageSpeak (message, nlpIntent) {
-    if (message.connect_state === 'disconnected' && message.a2dpstate !== 'close' &&
-       message.a2dpstate === 'opened') {
-      speakAndBackground(textTable['STRING_DISCONNECT_LINK'])
-    }
-    if (message.a2dpstate === 'opened' && message.connect_state === 'invalid' &&
-      message.play_state === 'invalid') {
-      if (message.linknum === 0) {
-        if (nlpIntent === 'connect_phone') {
-          speakAndBackground(textTable['STRING_CONNECT_MOBILE'] + textTable['STRING_OPEN_BEGIN'] +
-           nameToSpeak + textTable['STRING_CONNECT_MOBILE_END'])
-        } else {
-          if (wifi.getWifiState() !== wifi.WIFI_CONNECTED) {
-            mediaAndBackground('system://openbluetooth.ogg')
-          } else {
-            speakAndBackground(textTable['STRING_BROADCAST'] + textTable['STRING_OPEN_BEGIN'] +
-            nameToSpeak + textTable['STRING_OPEN_END'])
-          }
-        }
-      } else {
-        mediaAndSpeak(textTable['STRING_BROADCAST'], 'system://openbluetooth.ogg')
-      }
-    }
-    if (message.a2dpstate === 'open failed' && message.connect_state === 'invalid' &&
-    message.play_state === 'invalid') {
-      if (wifi.getWifiState() === wifi.WIFI_CONNECTED) {
-        activity.setForeground().then(() => {
-          speakAndExit(textTable['STRING_OPENFAILED'])
-        })
-      } else {
-        activity.setForeground().then(() => {
-          mediaAndExit('system://openbluetootherror.ogg')
-        })
-      }
-    }
-    if (message.a2dpstate === 'opened' && message.connect_state === 'connected' &&
-      message.play_state === 'invalid') {
-      mediaAndSpeak(textTable['STRING_CONNECED'] + message.connect_name,
-        'system://connectbluetooth.ogg')
+  function getAccessName (mode) {
+    if (mode === protocol.A2DP_MODE.SINK) {
+      return deviceName
+    } else {
+      return '若琪App'
```

Comment:
Could this string be moved to texts.json?

Reply:
Improved, please check again.